File: app/src/main/java/ru/goodibunakov/prodvtest/fragments/ChoseCityFragment.java
package ru.goodibunakov.prodvtest.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Objects;

import adapters.CityAdapter;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import ru.goodibunakov.prodvtest.R;
import ru.goodibunakov.prodvtest.utils.HawkHelper;

public class ChoseCityFragment extends Fragment {

    private Unbinder unbinder;

    @BindView(R.id.recycler)
    RecyclerView recyclerView;

    public ChoseCityFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_chose_city, container, false);
        unbinder = ButterKnife.bind(this, view);

        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        recyclerView.addItemDecoration(new DividerItemDecoration(Objects.requireNonNull(getActivity()), DividerItemDecoration.VERTICAL));
        CityAdapter adapter = new CityAdapter(HawkHelper.getInstance().checkIfContain(HawkHelper.ITEMS) ? HawkHelper.getInstance().getItem(HawkHelper.ITEMS) : new ArrayList<>());
        recyclerView.setAdapter(adapter);

        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/ru/goodibunakov/prodvtest/view/fragments/ChoseCityFragment.kt
package ru.goodibunakov.prodvtest.view.fragments


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.fragment_chose_city.*
import ru.goodibunakov.prodvtest.R
import ru.goodibunakov.prodvtest.adapters.CityAdapter
import ru.goodibunakov.prodvtest.utils.HawkHelper
import java.util.*


class ChoseCityFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_chose_city, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        recyclerView!!.setHasFixedSize(true)
        recyclerView!!.layoutManager = LinearLayoutManager(activity)
        recyclerView!!.addItemDecoration(DividerItemDecoration(Objects.requireNonNull(activity), DividerItemDecoration.VERTICAL))
        val adapter = CityAdapter(if (HawkHelper.getInstance().checkIfContain(HawkHelper.ITEMS)) HawkHelper.getInstance().getItem(HawkHelper.ITEMS) else ArrayList())
        recyclerView!!.adapter = adapter
    }
}